### PR TITLE
Add CentOS 8 support and remove CentOS 6

### PR DESCRIPTION
**Pull request by enrico-usai**
This patch is a rebase and cleanup of the following patches:

- Create packer file for Centos8 #749
- CentOS 8: Install python3 version of aws-cfn-bootstrap #750
- Add support for CentOS 8 in base_install and base_config logic #751
- Enable Torque compilation for CentOS8 #752
- CentOS 8: mark as unsupported OS for EFA and enable Intel MPI #754
- Centos8: Add Centos8 to pcluster_dcv_connect.sh #768
- CentOS 8: Add support for lustre and cw log agent to Centos8+ARM #777
- Centos 8: fix Intel HPC cache issue by setting `keepcache=True` #781
- CentOS 8: Fix bug in parallelcluster-ebsnvme-id #787
- Enable SGE compilation for CentOS8 #788
- Use version-1 superblock format for RAID on Centos8 #789

### CODE REVIEW

File: recipes/tests.rb [hunk 1]
```diff
@@ -290,20 +286,18 @@
...
+    bash 'check intel mpi version' do
+      cwd Chef::Config[:file_cache_path]
+      code <<-INTELMPI
+        set -e
+        # Initialize module
+        # Unset MODULEPATH to force search path reinitialization when loading profile
+        unset MODULEPATH
+        # Must execute this in a bash script because source is a bash built-in function
+        source /etc/profile.d/modules.sh
+        module load intelmpi && mpirun --help | grep 'Version 2019 Update 7'
```
Comment (rexcsn) [inline recipes/tests.rb:298]: This test should be checking for Intel MPI U8 with `grep 'Version 2019 Update 8'`

File: amis/packer_centos8.json [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        "curl --retry 3 -L -o /tmp/aws-cfn-bootstrap-py3-latest.tar.gz https://${bucket}/cloudformation-examples/aws-cfn-bootstrap-py3-latest.tar.gz",
+        "sudo dnf install -y python3-pip",
+        "sudo pip3 install /tmp/aws-cfn-bootstrap-py3-latest.tar.gz"
+      ]
+    },
+    {
+      "type" : "shell",
+      "inline" : [
+        "sudo rm -rf /tmp/* /var/tmp/* && sudo rm -f /root/.ssh/authorized_keys && rm -f ~/.ssh/authorized_keys",
+        "echo aws-parallelcluster-{{user `parallelcluster_version`}} | sudo tee /opt/parallelcluster/.bootstrapped"
```
Comment (rexcsn) [inline amis/packer_centos8.json:292]: Reminder to change to writing `aws-parallelcluster-cookbook-{{version}}` to bootstrap file

Comment (enrico-usai) [reply amis/packer_centos8.json]: Done!

File: recipes/tests.rb (hunk 1 above)
Comment (enrico-usai) [reply recipes/tests.rb]: Fixed, good catch, I removed it as part of the centos6 deprecation (and the rebase)

**Review by rexcsn — APPROVED** [review]
LGTM!🚀

File: libraries/helpers.rb [hunk 3]
```diff
@@ -277,16 +266,7 @@ def arm_instance?
...
-   node['platform'] == 'amazon',
-   node['platform'] == 'ubuntu'].any?
-end
-
-#
-# Check if the platform supports intel MPI
-#
-def platform_supports_impi?
-  [node['platform'] == 'centos' && node['platform_version'].to_i >= 7,
+  [node['platform'] == 'centos' && node['platform_version'].to_i < 8,
```
Comment (rexcsn) [inline libraries/helpers.rb:269]: Minor: Would be nice to add a `FIXME`

**Review by rexcsn — APPROVED** [review]
LGTM!

**Pull request merged**